### DISCUSSION TITLE
Contradiction between Energy and Force Accuracy in DPGEN Training

### DISCUSSION THREAD

**Discussion by zhangyu0104** [deepmodeling/dpgen#1692] [Q&A]
Dear all,

I am encountering an issue with the model accuracy while training iteratively using DPGEN for a Fe-H binary system, where the magnetism need to be considered. After several iterations, the accuracy ratio has improved from 70% to over 95%. However, when testing the model with `dp_test`, I noticed that while the root mean square error (RMSE) for force is relatively small, the RMSE for energy is significantly large.

![RMSE](https://github.com/user-attachments/assets/2d015a7b-43b4-471a-a353-3c1afc398aa9)

I am not sure whether this issue is related to the incorporation of magnetic effects. However, regardless of whether I fix the total magnetic moment or only set the initial magnetic moment, the energy RMSE continues to increase gradually with each iteration, while the force RMSE decreases. I also attempted to adjust the prefactors of energy and force during training, but this has not led to any improvements.

My input parameter file is attached below. I would greatly appreciate any insights or suggestions on how to address this issue.

[param.json](https://github.com/user-attachments/files/18206026/param.json)

Best regards,
Yu Zhang

**Comment by wanghan-iapcm** [comment]
Dear Yu,

The model trained during dpgen iterations pay little attention on the accuracy of energy. Note that we select configurations for labeling based on force criteria. 

For a production model, one needs a long-train and perhaps a few rounds of training (with --init-model option) with larger energy loss prefactors. 

Best,
Han
